### PR TITLE
Fixed missing domApi argument in recursive toVNode() call

### PR DESCRIPTION
**Pull request by katyo**

### CODE REVIEW

**Comment by staltz** [comment]
Looks good to me

**Comment by katyo** [comment]
I have no ideas how to test it.

**Comment by staltz** [comment]
Hints:

- There are tests for HTML DOM API: https://github.com/snabbdom/snabbdom/blob/master/test/htmldomapi.js
- There are tests for toVNode: https://github.com/snabbdom/snabbdom/blob/2271b7bdf15577eabd8de961f4e5bba5bd1515fe/test/core.js#L266

How about creating a fake `domApi` object where some functions have asserts inside them, then use toVNode in a way that will trigger the recursive calls.

File: test/core.js [hunk 1]
```diff
@@ -337,6 +337,42 @@ describe('snabbdom', function() {
...
         assert.strictEqual(elm.childNodes[0].textContent, 'Hello');
       })
+      it('can work with domApi', function () {
+        var domApi = {
+            tagName: function(elm) { return elm.tagName.toUpperCase(); },
+            getTextContent: function(elm) { return elm.text; },
+            isElement: function(elm) { return !!elm.tagName; },
+            isText: function(elm) { return !!elm.text; }
+        };
+        var Elm = function(tagName) {
```
Comment (caridy) [inline test/core.js]: I don't think u need `Elm` constructor, just create a DOM structure like described here: 

https://github.com/snabbdom/snabbdom/blob/master/test/core.js#L268-L273

Then call `toVNode()` on that structure and analyze the returned vnode structure to make sure that it is correct.

you can provide a domApi where tagName returns a custom tagname for any element processed, e.g.:

```js
tagName: function(elm) { return 'X-' + elm.tagName.toUpperCase(); },
```

which means that both, the parent a child vnodes should have a `sel` that contains the `x-` prefix.

**Comment by katyo** [comment]
It seems I amended last commit

**Comment by staltz** [comment]
Thank you @katyo !

**Pull request merged**